Pyspark/tests/test_regular_patients.py

```diff
@@ -0,0 +1,191 @@
+def consecutive_data(spark):
+    """Test dataframe data"""
+    data = [
+        (1, '2016-04-01', 0, 5),
+        (1, '2016-05-01', 1, 5),
+        (1, '2016-06-01', 2, 5),
+        (1, '2016-07-01', 3, 5),
+        (1, '2016-08-01', 4, 5),
+        (1, '2016-09-01', 5, 5),
+        (2, '2016-01-01', 0, 0),
+        (2, '2016-03-01', 0, 0),
+        (2, '2016-09-01', 0, 0),
+        (3, '2016-08-01', 0, 1),
+        (3, '2016-09-01', 1, 1),
+        (4, '2016-01-01', 0, 1),
+        (4, '2016-08-01', 0, 1),
+        (4, '2016-09-01', 1, 1)
+    ]
+    schema = StructType([
+    StructField('patient_id', StringType()),
+    StructField('effective_from_date_formatted', StringType()),
+    StructField('count_consec', LongType(), True),
+    StructField('max_count_consec', LongType(), True)
+    ])
+
+    df = spark.createDataFrame(data, schema)
+    df = df.withColumn('effective_from_date_formatted',
+                       f.to_date(f.col('effective_from_date_formatted')))
+    return df
+
+@pytest.fixture
+def final_result_data(spark):
+    """Test dataframe data"""
+    data = [
+        (1, True, False, False),
+        (2, False, False, False),
+        (3, False, False, False),
+        (4, False, False, False)
+        ]
+    schema = StructType([
+        StructField('patient_id', StringType()),
+        StructField('5months', BooleanType()),
+        StructField('9months', BooleanType()),
+        StructField('11months', BooleanType())
+        ])
+    df = spark.createDataFrame(data, schema)
+    return df
+
+
+@pytest.fixture
+def patients_data_instance(mocker, sample_data):
+    """Fixture to create a mock instance of PatientsData.
+    Returns: PatientsData- An instance of PatientsData with the mock loaded sample data"""
+    #mocking PatientData.load_data
+    mock_load_data = mocker.patch('regular_patients.regular_patients.PatientsData.load_data')
+    mock_load_data.return_value = sample_data
+    patients_data = PatientsData('sample-path')
+    return patients_data
+
+def test_format_date(patients_data_instance, formatted_date_data):
+    """Testing PatientsData.format_date()"""
+    result_df = (patients_data_instance.format_date()
+        .select('patient_id', 'effective_from_date_formatted'))
+    assertDataFrameEqual(result_df, formatted_date_data)
+
+def test_filter_by_date_range(patients_data_instance, filtered_data):
+    """Testing PatientsData.filter_by_date_range()"""
+    result_df = patients_data_instance.format_date()
+    result_df = patients_data_instance.filter_by_date_range(START_DATE, END_DATE)
+    result_df = result_df.select('patient_id', 'effective_from_date_formatted')
+    assertDataFrameEqual(result_df, filtered_data)
+    
+def test_checking_consecutives_records(patients_data_instance, consecutive_data):
+    """Testing PatientsData.checking_consecutives_records()"""
+    result_df = patients_data_instance.format_date()
+    result_df = patients_data_instance.filter_by_date_range(START_DATE, END_DATE)
```

Comment:
Also in your test you are setting `result_df = patients_data_instance.format_date()`  and on the next line reassigning it to the new value `result_df = patients_data_instance.filter_by_date_range(START_DATE, END_DATE)`
![image](https://github.com/user-attachments/assets/c5ef3ecd-3a48-471f-a3fb-6bab9e044a28)

In such case you are overwrite variable value